File: JavaScript/0080_remove_duplicates_from_sorted_array_II/solution.js
function remove_duplicates(nums) {
    let check_dict = {};
    for (let num of nums) {
        if (num in check_dict) {
            if (check_dict[num] < 2) {
                check_dict[num] ++;
            }
        } else {
            check_dict[num] = 1;
        }
    }
    let duplicate = [];
    for (let [key,value] of Object.entries(check_dict)) {
        for (let i = 0; i < value; i ++) {
            duplicate.push(key);
        }
    }
    let diff = nums.length - duplicate.length;
    for (let j = 0; j < diff; j ++) {
        duplicate.push(0);
    }
    for (let index = 0; index < nums.length; index ++) {
        nums[index] = duplicate[index];
    }
    return nums.length - diff;
}

nums = [1,1,1,2,2,3];
console.log(remove_duplicates(nums));

File: Python/0080_remove_duplicates_from_sorted_array_II/solution.py
def remove_duplicates(nums):
    check_dict = dict()
    for num in nums:
        if num in check_dict:
            if check_dict[num] < 2:
                check_dict[num] += 1
        else:
            check_dict[num] = 1
    duplicate = list()
    for key,value in check_dict.items():
        for _ in range(value):
            duplicate.append(key)
    diff = len(nums) - len(duplicate)
    for _ in range(diff):
        duplicate.append(0)
    for index in range(len(nums)):
        nums[index] = duplicate[index]
    return len(nums) - diff

nums = [1,1,1,2,2,3]
print(remove_duplicates(nums))